engine/robot_state.py

```diff
@@ -0,0 +1,138 @@
+    import electrical.gps as GPS 
+    import electrical.imu as IMU 
+    import electrical.radio_module as RadioModule
+    import serial
+    # from engine.sensor_module import SensorModule # IMU + GPS testing
+
+class Robot_State:
+    """
+    A class containing robot-specific information about the current state of a robot.
+
+    Initializes the information for the robot.
+    Parameters:
+        x_pos: the x position of the robot, where (0,0) is the bottom left corner of the grid with which the Robot's related Mission was initialized
+        y_pos: the y position of the robot where (0,0) is the bottom left corner of the grid with which the Robot's related Mission was initialized
+        heading: the theta of the robot in radians, where North on the grid is equal to 0.
+        epsilon: dictates the turning radius of the robot. Lower epsilon results in tighter turning radius.
+        max_velocity: the maximum velocity of the robot
+        radius: the radius of the robot
+    Preconditions:
+        x_pos is an int within the bounds of the user-defined traversal grid
+        y_pos is an int within the bounds of the user-defined traversal grid
+        heading is a float
+        epsilon is a float
+        max_velocity is a float
+        radius is a float
+    """
+    def __init__(self, x_pos, y_pos, heading, epsilon, max_velocity, radius):
+        """
+        Instance Attributes:
+            epsilon: dictates the turning radius of the robot. Lower epsilon results in tighter turning radius.
+            max_velocity: the maximum velocity of the robot
+            radius: the radius of the robot
+            width: width of the robot in cm
+            front_ultrasonic: the ultrasonic at the front of the robot, used for detecting obstacles
+            lf_ultrasonic: the ultrasonic at the front of the left side of the robot, used for boundary following
+            lb_ultrasonic: the ultrasonic at the back of the left side of the robot, used for boundary following
+            rf_ultrasonic: the ultrasonic at the front of the right side of the robot, used for boundary following
+            rb_ultrasonic: the ultrasonic at the back of the right side of the robot, used for boundary following
+            is_sim: False if the physical robot is being used, True otherwise
+            is_store: False if csv data should not be stored, True otherwise
+            position_kp: the proportional factor of the position PID
+            position_ki: the integral factor of the position PID
+            position_kd: the derivative factor of the position PID
+            position_noise: the flat amount of noise added to the robot's phase on each localization step
+            heading_kp: the proportional factor of the heading PID
+            heading_ki: the integral factor of the heading PID
+            heading_kd: the derivative factor of the heading PID
+            heading_noise: ?
```

Comment:
we should probably just remove  this?

Reply:
same as above -- but also does this mean there's no heading pid?

Reply:
I think Julie is talking about the heading_noises?